indexer/ai/src/main/java/com/google/cloudsearch/ai/StandardSkillSentimentExtraction.java

```diff
@@ -0,0 +1,271 @@
+            }
+        }
+    }
+
+    /**
+     * Returns the filters
+     * @return  Returns the filters in JSON Format.
+     */
+    @Override
+    public JSONObject getFilter() {
+        JSONObject obj = new JSONObject();
+        obj.put(Constants.CONFIG_SENTIMENT_SCORE_POSITIVE, this.sentimentScorePositive);
+        obj.put(Constants.CONFIG_SENTIMENT_SCORE_NEGATIVE, this.sentimentScoreNegative);
+        obj.put(Constants.CONFIG_SENTIMENT_MAGNITUDE_THRESHOLD, this.sentimentMagnitudeThreshold);
+        obj.put(Constants.CONFIG_SENTIMENT_MAGNITUDE_IGNORE, this.sentimentMagnitudeIgnore);
+        return obj;
+    }
+
+    /**
+     * Parses the JSON Object containing the filters for the skill and validate it before calling the setter.
+     * @param filter  JSON Object containing the filters
+     * @throws InvalidConfigException     Throws exception if the filter name is not supported.
+     */
+    @Override
+    public void parseFilter(JSONObject filter) throws InvalidConfigException {
+        if(filter == null){
+            return;
+        }
+        setFilter(filter);
+        //Check validity of filter
+        if(Math.abs(this.sentimentScorePositive) > 1.0) {
+            throw new InvalidConfigException("Filter sentimentScorePositive cannot be greater than 1.0 or less than -1.0");
+        }
+        if(Math.abs(this.sentimentScoreNegative) > 1.0){
+            throw new InvalidConfigException("Filter sentimentScoreNegative cannot be greater than 1.0 or less than -1.0");
+        }
+        if( !(this.sentimentMagnitudeIgnore.equals("yes") || this.sentimentMagnitudeIgnore.equals("no"))){
+            throw new InvalidConfigException("Filter sentimentMagnitudeIgnore can only be either yes or no.");
+        }
+        if(this.sentimentScorePositive < this.sentimentScoreNegative) {
+            throw new InvalidConfigException("Filter sentimentScorePositive cannot be smaller than sentimentScoreNegative.");
+        }
+    }
+
+    /**
+     * Find out the sentiment based on the filters from the given sentiment score and magnitude.
+     * @param sentimentScore        Sentiment Score of the document obtained from the API
+     * @param sentimentMagnitude    Sentiment Magnitude of the document obtained from the API
+     * @return                      Returns the inferred sentiment based on the filter. The returned value can
+     *                              be Positive, Negative, Neutral or Mixed.
+     */
+    private String getSentiment(Double sentimentScore, Double sentimentMagnitude) {
+        if(this.sentimentMagnitudeIgnore.equals("yes")){
+            if( sentimentScore >= this.sentimentScorePositive){
+                return Constants.CONFIG_SENTIMENT_POSITIVE;
+            }
+            else if(sentimentScore <= this.sentimentScoreNegative){
+                return Constants.CONFIG_SENTIMENT_NEGATIVE;
+            }
+            else{
+                return Constants.CONFIG_SENTIMENT_NEUTRAL;
+            }
```

Comment:
All the 3 if, else and else (starting at lines 163, 174 and 187)share this same code. We can create a function and call it for all three?

Reply:
Done I refactored this.